### ISSUE TITLE
parallel build hangs

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1640]
Hi,
it seems that if I build some docs with Sphinx 1.2.3 on Linux 64 on two or more cores (no matter which is the target format), then the process hangs for ever. When I interrupt it I get the following:

```
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
^Citing output... [  5&#37;] reference/ConicVariable_toString .. reference/Constraint_dual                                                                                                                                                                                                                                         
interrupted!
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/remote/public/linux/64-x86/python/3.4.1/lib/python3.4/multiprocessing/popen_fork.py", line 30, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
```

This does not happens if I switch off the output.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1640
- Originally reported by: Andrea Cassioli
- Originally created at: 2014-12-03T15:25:23.088

**Comment by birkenfeld** [comment]
Hmm. Does that only happen with a specific project for you, or for all projects?

**Comment by shimizukawa** [comment]
@andreacassioli can you take a look this issue?

**Comment by andreacassioli** [comment]
Hi,
sorry for the delay.

Actually I cannot reproduce the issue anymore. My project has changed considerably and my sys admin could have reinstalled Sphinx from scratch.

**Comment by birkenfeld** [comment]
Thanks for letting us know.

**Comment by andreacassioli** [comment]
Hi, the issue seems to be still there.

I am running Sphinx 1.2b3 on Linux 64. 

On a specific machine, it happens every time I set the number of jobs greater than one. Sphinx hangs while writing, i.e. during the second phase. After some warnings the process just hangs and I have to kill it. The very same code works fine on other machines.

I cannot install a beta version now, as the system that hangs is our internal build system and it runs also other processes.

**Comment by birkenfeld** [comment]
I assume you mean 1.3b3? It would be interesting to know if the machine in question can run any other multiprocessing jobs fine.

**Comment by andreacassioli** [comment]
The machine can definitely run multiprocess jobs. The version is actually 1.2.2.

**Comment by andreacassioli** [comment]
It seems that reducing the output makes the trick. I included option -q and now it builds on multithreads.

**Comment by birkenfeld** [comment]
For 1.2.2 the parallel implementation is quite different. It is possible that there are bugs left there that are now fixed for 1.3.

**Comment by andreacassioli** [comment]
Hi,
I am still experiencing the issue in a quite predictable way. The build hangs during the parallel write phase. Even with just to jobs, it just stop and the threads hang forever.
Funny, when I kill the build one of the thread dies, the other don't, but instead stays alive forever. When I kill it I get:

```
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
^Citing output... [  9%] dataio-param .. dparam-data-tol-aij-large                                                                                                                                                                                                                                                             
interrupted!
^CError in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/lib/python3.4/multiprocessing/popen_fork.py", line 30, in poll
    pid, sts = os.waitpid(self.pid, flag)
KeyboardInterrupt
```

Sorry, I cannot make a small example. But I have noticed it only happens when I build large projects. In particular, it seems related to output on screen. If I keep Sphinx silent, it is fine. Moreover, it happens both with html and Latex.

I am running on Sphinx 1.3.1 installed via PIP.
